custom_components/sector/binary_sensor.py

```diff
@@ -40,76 +76,37 @@ async def async_setup_entry(
-                    BinarySensorDeviceClass.DOOR,
-                    device_model,
-                )
-            )
-        if "low_battery" in sensors:
-            _LOGGER.debug(
-                "Adding battery to %s as  model '%s' with type '%s'",
-                serial_no,
-                device_model,
-                device_type,
-            )
-            entities.append(
-                SectorAlarmBinarySensor(
-                    coordinator,
-                    serial_no,
-                    "low_battery",
-                    device,
-                    BinarySensorDeviceClass.BATTERY,
-                    device_model,
+        # Add each sensor entity based on available descriptions
+        for description in SENSOR_DESCRIPTIONS:
+            if description.sensor_type in sensors:
+                entities.append(
+                    SectorAlarmBinarySensor(
+                        coordinator,
+                        serial_no,
+                        device,
+                        description,
+                        device_model,
+                    )
                 )
-            )
-        else:
-            _LOGGER.warning(
-                "No low_battery sensor found for device %s (%s). Confirmed sensors: %s",
-                serial_no,
-                device_model,
-                sensors,
-            )
-        if "leak_detected" in sensors:
-            entities.append(
-                SectorAlarmBinarySensor(
-                    coordinator,
-                    serial_no,
-                    "leak_detected",
-                    device,
-                    BinarySensorDeviceClass.MOISTURE,
-                    device_model,
-                )
-            )
-        if "alarm" in sensors:
-            entities.append(
-                SectorAlarmBinarySensor(
-                    coordinator,
-                    serial_no,
-                    "alarm",
-                    device,
-                    BinarySensorDeviceClass.SAFETY,
-                    device_model,
-                )
-            )
 
-    # Add panel online status sensor
+    # Handle panel online status as another entity in the setup
     panel_status = coordinator.data.get("panel_status", {})
     panel_id = entry.data.get("panel_id")
     serial_no = panel_status.get("SerialNo") or panel_id
-    entities.append(
-        SectorAlarmPanelOnlineBinarySensor(
-            coordinator,
-            serial_no,
-            "online",
-            BinarySensorDeviceClass.CONNECTIVITY,
+    if serial_no:  # Add panel status sensor if serial number is available
+        entities.append(
+            SectorAlarmBinarySensor(
+                coordinator,
+                serial_no,
+                {"name": "Sector Alarm Panel"},
+                next(
+                    desc
+                    for desc in SENSOR_DESCRIPTIONS
+                    if desc.sensor_type == "online"
+                ),
```

Comment:
Instead of having this in `SENSOR_DESCRIPTIONS` I suggest you make an entity description here directly (and then it can be removed from `SENSOR_DESCRIPTIONS`).

Reply:
Done.

Reply:
Not fixed as you still have `next(...)` here.